### PR TITLE
docs: Specify the instruction format on little-endian machine.

### PR DESCRIPTION
**Pull request by oujunli**
**Summary of changes**:

Fixes: #issue-number

```release-note
<!-- Enter the release note text here if needed -->
```

**How to test (optional)**:
e.g. Run `make tests`, `make runtime-tests` and/or some more specific way to
exercise the new behavior(s).
Note: [The runtime testsuite will modify the host it runs on.](http://cilium.readthedocs.io/en/latest/contributing/#testsuite)

### CODE REVIEW

**Review by borkmann — COMMENTED** [review]
Thanks a lot for the patch Junli! Could you elaborate, are you referring here specifically when on little endian? I've been using the order as in struct bpf_insn:

```
struct bpf_insn {
	__u8	code;		/* opcode */
	__u8	dst_reg:4;	/* dest register */
	__u8	src_reg:4;	/* source register */
	__s16	off;		/* signed offset */
	__s32	imm;		/* signed immediate constant */
};
```

**Comment by oujunli** [comment]
yes, test in a VM (Ubuntu-16.04.01 x86_64), is little endian, but debug [bcc](https://github.com/iovisor/bcc) helloworld.py example, the ebpf assemble code:
```
# ./hello_world.py  
0: (b7) r1 = 2593
1: (6b) *(u16 *)(r10 -4) = r1
2: (b7) r1 = 1684828783
3: (63) *(u32 *)(r10 -8) = r1
4: (18) r1 = 0x57202c6f6c6c6548
6: (7b) *(u64 *)(r10 -16) = r1
7: (b7) r1 = 0
8: (73) *(u8 *)(r10 -2) = r1
9: (bf) r1 = r10
10: (07) r1 += -16
11: (b7) r2 = 15
12: (85) call bpf_trace_printk#6
13: (b7) r0 = 0
14: (95) exit
processed 14 insns, stack depth 16
...
```
break in kernel ___bpf_prog_run:
```
(gdb) x/15xg insn
0xffffc9000049d030:	0x00000a21000001b7	0x00000000fffc1a6b
0xffffc9000049d040:	0x646c726f000001b7	0x00000000fff81a63
0xffffc9000049d050:	0x6c6c654800000118	0x57202c6f00000000
0xffffc9000049d060:	0x00000000fff01a7b	0x00000000000001b7
0xffffc9000049d070:	0x00000000fffe1a73	0x000000000000a1bf
0xffffc9000049d080:	0xfffffff000000107	0x0000000f000002b7
0xffffc9000049d090:	0xffff569000000085	0x00000000000000b7
0xffffc9000049d0a0:	0x0000000000000095
```
the bpf instruction bit order is the opposite of the document, also read the [bpf-doc](https://github.com/iovisor/bpf-docs/blob/master/eBPF.md) confirm that, miss something else?
Thanks.

**Comment by borkmann** [comment]
> the bpf instruction bit order is the opposite of the document, also read the bpf-doc confirm that, miss something else?

The document assumes little-endian as well, whereas here big-endian. I'm fine either way but we should definitely in the commit specify which encoding is used here to not confuse readers. Why not just mention big endian encoding which seems most natural for a human to read the doc at least and fits with `struct bpf_insn` order I mentioned earlier.

**Comment by oujunli** [comment]
yes, you are right, I updated the PR, Thanks.

File: Documentation/bpf.rst [hunk 1]
```diff
@@ -186,11 +186,11 @@ The instruction format is modeled as two operand instructions, which helps mappi
 BPF instructions to native instructions during JIT phase. The instruction set is
 of fixed size, meaning every instruction has 64 bit encoding. Currently, 87 instructions
 have been implemented and the encoding also allows to extend the set with further
-instructions when needed. The instruction encoding of a single 64 bit instruction is
-defined as a bit sequence from most significant bit (MSB) to least significant bit (LSB)
-of ``op:8``, ``dst_reg:4``, ``src_reg:4``, ``off:16``, ``imm:32``. ``off`` and
-``imm`` is of signed type. The encodings are part of the kernel headers and
-defined in ``linux/bpf.h`` header, which also includes ``linux/bpf_common.h``.
+instructions when needed. The instruction encoding of a single 64 bit instruction on a
+little-endian machine is defined as a bit sequence from most significant bit (MSB) to
```
Comment (borkmann) [inline Documentation/bpf.rst]: This would need to say big-endian here if we keep the order as is.

Comment (oujunli) [reply Documentation/bpf.rst]: Sorry for my mistake, updated.

**Comment by borkmann** [comment]
Merged, thanks a lot Junli!

**Pull request merged**